plugins/scaffolder/src/api.ts

```diff
@@ -145,25 +150,14 @@ export class ScaffolderClient implements ScaffolderApi {
     return schema;
   }
 
-  /**
-   * Executes the scaffolding of a component, given a template and its
-   * parameter values.
-   *
-   * @param templateName Template name for the scaffolder to use. New project is going to be created out of this template.
-   * @param values Parameters for the template, e.g. name, description
-   */
   async scaffold(
     templateName: string,
     values: Record<string, any>,
   ): Promise<string> {
-    const token = await this.identityApi.getIdToken();
     const url = `${await this.discoveryApi.getBaseUrl('scaffolder')}/v2/tasks`;
```

Comment:
Thoughts on using `plugin://scaffolder/v2/tasks` here instead?

Reply:
I was a little bit hesitant, and thought maybe we should talk about it first (possibly to be leveraged in a separate PR rather than now).

As soon as we start relying on these two middlewares existing, they become first class and effectively required. Their semantics become engrained into the different plugins. And the plugin resolution is after all a bit more of a convenient sugar than the identity/auth injection, so of less immediate value.

I could totally start using it in theory of course (and making sure that the mock api also supports that - with the implications that come with that, since it in turn uses the discovery api which now comes into play too!). But thought it prudent to do separately

Reply:
Yep, we'd have to update the `FetchApi` to say that you `MUST` support `plugin://`. Was thinking it's worth the debate here since it'd be yet another breaking change to introduce that in the future too, although at least that has a deprecation path. If we don't want to use `plugin://` we should consider whether it should exist at all, it might be nice as an internal utility still but it would be a shame if non-private plugins can't use it.

I think a possible way forward for now could be to provide a simple way to wrap a `FetchApi` to have it support `plugin://`, which I guess we do since the `FetchApi` implements `fetch` so we can just add another layer to it 😅 
We could then start using internally within plugins, so the `ScaffolderClient` would wrap up the `FetchApi` that it receives in the constructor to have it support `plugin://` internally. That way we can start using it and get a bit more of a feel for if it's the way we wanna go.

Would also say that it's a slightly urgent decision since adding the requirement to support `plugin://` is a breaking change x)

Reply:
Part of the reason I wanted the api to be of the shape `{ fetch }` is actually because I thought we'd want it to actually become `{ fetch, with }` or similar.

Then the constructor could do `this.fetchApi = options.fetchApi.with(FetchMiddlewares.protocolTranslation...)` :)

Oooor we just make a wrapper class so we don't need to affect the shape of the FetchApi type. But you know I like chaining.